src/main/java/org/spongepowered/api/entity/ai/Goal.java

```diff
@@ -64,15 +64,15 @@
      * @param task The task to run
      * @return This goal, for chaining
      */
-    Goal<O> addTask(int priority, AITask<? extends O> task);
+    Goal<O> addTask(int priority, AITask<? super O> task);
```

Comment:
@Zidane I think you handled these generics incorrectly. Any thoughts on these? Also pinging @kashike to pick out issues.